lib/fog/key_manager/openstack.rb

```diff
@@ -0,0 +1,103 @@
+                 :openstack_project_domain, :openstack_user_domain, :openstack_domain_name,
+                 :openstack_project_domain_id, :openstack_user_domain_id, :openstack_domain_id,
+                 :openstack_identity_prefix, :openstack_temp_url_key, :openstack_cache_ttl
+
+
+      ## MODELS
+      #
+      model_path 'fog/key_manager/openstack/models'
+      model       :secret
+      collection  :secrets
+      model       :container
+      collection  :containers
+
+      ## REQUESTS
+
+      # secrets
+      request_path 'fog/key_manager/openstack/requests'
+      request :create_secret
+      request :list_secrets
+      request :get_secret
+      request :get_secret_payload
+      request :get_secret_metadata
+      request :delete_secret
+
+      # containers
+      request :create_container
+      request :get_container
+      request :list_containers
+      request :delete_container
+
+      class Real
+        include Fog::OpenStack::Core
+
+        def self.not_found_class
+          Fog::KeyManager::OpenStack::NotFound
+        end
+
+        def initialize(options = {})
+          initialize_identity options
+
+          @openstack_service_type           = options[:openstack_service_type] || ['key-manager']
+          @openstack_service_name           = options[:openstack_service_name]
+          @connection_options               = options[:connection_options] || {}
+
+          authenticate
+          set_api_path
+
+          @persistent = options[:persistent] || false
+          @connection = Fog::Core::Connection.new("#{@scheme}://#{@host}:#{@port}", @persistent, @connection_options)
+        end
+
+        def set_api_path
+          @path.sub!(%r{/$}, '')
+          unless @path.match(SUPPORTED_VERSIONS)
+            @path = supported_version(SUPPORTED_VERSIONS, @openstack_management_uri, @auth_token, @connection_options)
+          end
+        end
+
+        def supported_version(supported_versions, uri, auth_token, connection_options = {})
+          connection = Fog::Core::Connection.new("#{uri.scheme}://#{uri.host}:#{uri.port}", false, connection_options)
+          response = connection.request({ :expects => [200, 204, 300],
+                                          :headers => {'Content-Type' => 'application/json',
+                                                       'Accept' => 'application/json',
+                                                       'X-Auth-Token' => auth_token},
+                                          :method => 'GET'
+                                        })
+
+          body = Fog::JSON.decode(response.body)
+          version = nil
+
+          versions =  body.fetch('versions',{}).fetch('values',[])
+          versions.each do |v|
+            if v.fetch('id', "").match(supported_versions) &&
+              (v.fetch('status','').downcase == 'current' || v.fetch('status','').downcase == 'supported' || v.fetch('status','').downcase == 'stable')
```

Comment:
Sure! would be nice to not have an own method to check the version. We should extend the existing one.